### ISSUE TITLE
Is it possible to make the output image original size?

### ISSUE THREAD

**Issue by MultiPath** [gradio-app/gradio#287]
Hi, I am developing a demo using gradio for GAN output and the typical image size is around 1024. 
But the results looks pretty small right now. I think it automatically resize the image when showing in browser.

Thanks

**Comment by MultiPath** [comment]
This does not see work for the latest version?
I tried to update my demo to 2.8.13. The output image is super small.

![image](https://user-images.githubusercontent.com/5780274/159999914-1b529b4e-2259-4827-aa3b-47d1bb5f559c.png)

**Comment by MultiPath** [comment]
It seems working in this way. Thanks a lot

**Comment by oms12** [comment]
> @MultiPath did you try this [#60 (comment)](https://github.com/gradio-app/gradio/issues/60#issuecomment-937968278)
> 
> so for example where you can change the values in the css
> 
> css = ".output_image {height: 40rem !important; width: 100% !important;}"
> 
> iface = gr.Interface(gan_inference, inputs=gr.inputs.Image(type="pil"), outputs=gr.outputs.Image(type="pil"), title=title, description=description, article=article, examples=examples, css=css, enable_queue=True) iface.launch(debug=True)

```
def sample(sentence):
    sentences = [sentence]
    batch_size, device = 1, 'cpu'
    truncation, trunc_rate = True, 0.88
    z_dim = 100
    vocab_size, wordtoix = build_word_dict(pickle_path)
    captions, cap_lens, _ = tokenize(wordtoix, sentences)
    sent_embs, _  = prepare_sample_data(captions, cap_lens, text_encoder, device)
    caption_num = sent_embs.size(0)
    # get noise
    if truncation==True:
        noise = truncated_noise(batch_size, z_dim, trunc_rate)
        noise = torch.tensor(noise, dtype=torch.float).to(device)
    else:
        noise = torch.randn(batch_size, z_dim).to(device)
    # sampling
    with torch.no_grad():
        fakes = []
        for i in range(caption_num):
            sent_emb = sent_embs[i].unsqueeze(0).repeat(batch_size, 1)
            fakes = netG(noise, sent_emb)
        img = fakes[0]
        im = img.data.cpu().numpy()
        # [-1, 1] --> [0, 255]
        im = (im + 1.0) * 127.5
        im = im.astype(np.uint8)
        print(im.shape)
        im = np.transpose(im, (1, 2, 0))
        im = Image.fromarray(im)
        return im
    # return "gautam"
title = "DF-GAN"
description = "Generation of the image"
article = "For Generation of the image"
css = ".output-image, .input-image, .image-preview {height: 600px !important}"
# Create the Gradio demo
demo = gr.Interface(fn=sample, # mapping function from input to output
                    inputs="text", # what are the inputs?
                    outputs=gr.outputs.Image(type="numpy", label="annotated document"), # our fn has two outputs, therefore we have two outputs
                    title=title,
                    description=description,
                    css=css,
                    article=article,
                    enable_queue=True)

# Launch the demo!
demo.launch(debug=False, # print errors locally?
            share=True) # generate a publically shareable URL?
```

I am getting the image in the output box but it is not resized according to my css. Please suggest something.

**Comment by Ol0fmeistR** [comment]
> > @MultiPath did you try this [#60 (comment)](https://github.com/gradio-app/gradio/issues/60#issuecomment-937968278)
> > so for example where you can change the values in the css
> > css = ".output_image {height: 40rem !important; width: 100% !important;}"
> > iface = gr.Interface(gan_inference, inputs=gr.inputs.Image(type="pil"), outputs=gr.outputs.Image(type="pil"), title=title, description=description, article=article, examples=examples, css=css, enable_queue=True) iface.launch(debug=True)
> 
> ```
> def sample(sentence):
>     sentences = [sentence]
>     batch_size, device = 1, 'cpu'
>     truncation, trunc_rate = True, 0.88
>     z_dim = 100
>     vocab_size, wordtoix = build_word_dict(pickle_path)
>     captions, cap_lens, _ = tokenize(wordtoix, sentences)
>     sent_embs, _  = prepare_sample_data(captions, cap_lens, text_encoder, device)
>     caption_num = sent_embs.size(0)
>     # get noise
>     if truncation==True:
>         noise = truncated_noise(batch_size, z_dim, trunc_rate)
>         noise = torch.tensor(noise, dtype=torch.float).to(device)
>     else:
>         noise = torch.randn(batch_size, z_dim).to(device)
>     # sampling
>     with torch.no_grad():
>         fakes = []
>         for i in range(caption_num):
>             sent_emb = sent_embs[i].unsqueeze(0).repeat(batch_size, 1)
>             fakes = netG(noise, sent_emb)
>         img = fakes[0]
>         im = img.data.cpu().numpy()
>         # [-1, 1] --> [0, 255]
>         im = (im + 1.0) * 127.5
>         im = im.astype(np.uint8)
>         print(im.shape)
>         im = np.transpose(im, (1, 2, 0))
>         im = Image.fromarray(im)
>         return im
>     # return "gautam"
> title = "DF-GAN"
> description = "Generation of the image"
> article = "For Generation of the image"
> css = ".output-image, .input-image, .image-preview {height: 600px !important}"
> # Create the Gradio demo
> demo = gr.Interface(fn=sample, # mapping function from input to output
>                     inputs="text", # what are the inputs?
>                     outputs=gr.outputs.Image(type="numpy", label="annotated document"), # our fn has two outputs, therefore we have two outputs
>                     title=title,
>                     description=description,
>                     css=css,
>                     article=article,
>                     enable_queue=True)
> 
> # Launch the demo!
> demo.launch(debug=False, # print errors locally?
>             share=True) # generate a publically shareable URL?
> ```
> 
> I am getting the image in the output box but it is not resized according to my css. Please suggest something.

If you're talking about resizing the output box then maybe try something like this: 

`outputs=gr.Image(type='type', label='label').style(height=400, width=400)`

**Comment by zero14533** [comment]
How can I get the input and output boxes to be displayed at the bottom in gradio

**Comment by freddyaboulton** [comment]
Not sure what you mean @zero14533 . Do you mean on the same row? Can you please elaborate?

**Comment by abhishek0093** [comment]
> > > @MultiPath did you try this [#60 (comment)](https://github.com/gradio-app/gradio/issues/60#issuecomment-937968278)
> > > so for example where you can change the values in the css
> > > css = ".output_image {height: 40rem !important; width: 100% !important;}"
> > > iface = gr.Interface(gan_inference, inputs=gr.inputs.Image(type="pil"), outputs=gr.outputs.Image(type="pil"), title=title, description=description, article=article, examples=examples, css=css, enable_queue=True) iface.launch(debug=True)
> > 
> > 
> > ```
> > def sample(sentence):
> >     sentences = [sentence]
> >     batch_size, device = 1, 'cpu'
> >     truncation, trunc_rate = True, 0.88
> >     z_dim = 100
> >     vocab_size, wordtoix = build_word_dict(pickle_path)
> >     captions, cap_lens, _ = tokenize(wordtoix, sentences)
> >     sent_embs, _  = prepare_sample_data(captions, cap_lens, text_encoder, device)
> >     caption_num = sent_embs.size(0)
> >     # get noise
> >     if truncation==True:
> >         noise = truncated_noise(batch_size, z_dim, trunc_rate)
> >         noise = torch.tensor(noise, dtype=torch.float).to(device)
> >     else:
> >         noise = torch.randn(batch_size, z_dim).to(device)
> >     # sampling
> >     with torch.no_grad():
> >         fakes = []
> >         for i in range(caption_num):
> >             sent_emb = sent_embs[i].unsqueeze(0).repeat(batch_size, 1)
> >             fakes = netG(noise, sent_emb)
> >         img = fakes[0]
> >         im = img.data.cpu().numpy()
> >         # [-1, 1] --> [0, 255]
> >         im = (im + 1.0) * 127.5
> >         im = im.astype(np.uint8)
> >         print(im.shape)
> >         im = np.transpose(im, (1, 2, 0))
> >         im = Image.fromarray(im)
> >         return im
> >     # return "gautam"
> > title = "DF-GAN"
> > description = "Generation of the image"
> > article = "For Generation of the image"
> > css = ".output-image, .input-image, .image-preview {height: 600px !important}"
> > # Create the Gradio demo
> > demo = gr.Interface(fn=sample, # mapping function from input to output
> >                     inputs="text", # what are the inputs?
> >                     outputs=gr.outputs.Image(type="numpy", label="annotated document"), # our fn has two outputs, therefore we have two outputs
> >                     title=title,
> >                     description=description,
> >                     css=css,
> >                     article=article,
> >                     enable_queue=True)
> > 
> > # Launch the demo!
> > demo.launch(debug=False, # print errors locally?
> >             share=True) # generate a publically shareable URL?
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I am getting the image in the output box but it is not resized according to my css. Please suggest something.
> 
> If you're talking about resizing the output box then maybe try something like this:
> 
> `outputs=gr.Image(type='type', label='label').style(height=400, width=400)`

Is there a way to achieve the same using gradio Blocks instead of interface ?

**Comment by PaulWongDlut** [comment]
> For those looking to achieve the same using`gr.Blocks()`instead of `gr.Interface()`, you can specify css in gr.Blocks() and the same would be applied to entire block. Set all the relevant images to same class using `elem_classes` in gr.Images and target it through css something like discussed above `(css = ".output-image, .input-image, .image-preview {height: 600px !important}")`

this doesn't work for me under gradio4.38.1
